config/configtls/configtls.go

```diff
@@ -178,7 +178,7 @@ func (c TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
 
 func convertVersion(v string) (uint16, error) {
 	if v == "" {
-		return 0, nil // default
+		return tls.VersionTLS12, nil // default
```

Comment:
Breaking change in the CHANGELOG?

Reply:
Is this a breaking change, though? I wouldn't expect any OpenTelemetry Collector setup to break because of this change. Only setups using very old firewalls/load balancers would break, causing the operator to have the need to specify a lower version.

Reply:
If folks relied on TLS < 1.2, then upgrading to this new collector would break their deployments.

Reply:
> relied on TLS

Right, but this is the part that I find unlikely: who's relying on TLS nowadays? I can only think of legacy applications that were built a long time ago when 1.2 wasn't available as part of the underlying TLS libraries (openssl and alike). Everything around it, like routers, proxies, and load balancers likely support 1.2 already. Given that the collector is the piece that sits behind a proxy/router/load balancer, and we know it supports 1.2, I find it highly unlikely we'll have problems.

If you think it's worth being extra-cautious, I can send in a PR that moves this entry to the breaking change part.